x/evm/keeper/grpc_query.go

```diff
@@ -276,6 +276,34 @@ func (q Querier) Event(c context.Context, req *types.EventRequest) (*types.Event
 	return &types.EventResponse{Event: &event}, nil
 }
 
+// DepositState returns the status of the deposit matching the given chain, tx ID and burner address
+// Deprecated
+func (q Querier) DepositState(c context.Context, req *types.DepositStateRequest) (*types.DepositStateResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	ck, err := q.keeper.ForChain(ctx, req.Chain)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	if _, status, ok := ck.GetDepositByTxIDBurnAddr(ctx, req.Params.TxID, req.Params.BurnerAddress); ok {
+		return &types.DepositStateResponse{Status: status}, nil
+	}
+
+	hasSameBurnerAddress := func(deposit types.ERC20Deposit) bool {
+		return deposit.BurnerAddress == req.Params.BurnerAddress
+	}
+
+	// we can only return the first matching deposit at this point despite the fact that there might be many
+	if slices.Any(funcs.Must(ck.GetDepositsByTxID(ctx, req.Params.TxID, types.DepositStatus_Confirmed)), hasSameBurnerAddress) {
```

Comment:
there is an edge case deposits are partially burned, but doesn't matter that much

Reply:
That's right it doesn't matter. The query currently is only used by the recovery script to avoid confirming the same deposits twice.